src/main/java/io/reactivex/observers/TestObserver.java

```diff
@@ -678,6 +751,65 @@ public final boolean awaitTerminalEvent(long duration, TimeUnit unit) {
     }
 
     /**
+     * Sets the initial fusion mode if the upstream supports fusion.
+     * @param mode the mode to establish, see the {@link QueueDisposable} constants
+     * @return this
+     */
+    public final TestObserver<T> setInitialFusionMode(int mode) {
```

Comment:
Yeah I'm torn here. I understand the convenience offered but I worry
because it's only serving the built-in tests and not any consumers.

I will say that your to() proposal affords a lot of other benefits. It
eliminates the need for extend() (that I think was talked about recently in
an issue), has a name matching toXxx() for parity with the built-in
conversions methods, and allows the use of other custom test observers
(which we tend to use more than built-in TestObserver) with the same
support as the built-in one.

On Mon, Aug 22, 2016, 5:33 PM David Karnok <notifications@github.com> wrote:

> In src/main/java/io/reactivex/observers/TestObserver.java
> <https://github.com/ReactiveX/RxJava/pull/4403#discussion_r75763264>:
>
> > @@ -678,6 +751,65 @@ public final boolean awaitTerminalEvent(long duration, TimeUnit unit) {
> >      }
> >
> >      /**
> > +     * Sets the initial fusion mode if the upstream supports fusion.
> > +     * @param mode the mode to establish, see the {@link QueueDisposable} constants
> > +     * @return this
> > +     */
> > +    public final TestObserver<T> setInitialFusionMode(int mode) {
>
> The problem with that is that Observable.test() would need to return it
> or one would need to cast it to the internal implementation. We can move
> the constants into the public space. The other option would be to move test
> consumers into the test source set that would remove the test operators
> as well. Damn lack of extension methods.
>
> Also a less ugly version would be to turn test() methods into calls of
> to(Function) and provide TestHelper.newTest(int fusionMode)that return
> some customTestObserver`.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ReactiveX/RxJava/pull/4403/files/7cdb19eedab11edfb2af7afcbf3139939493d01a#r75763264>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEEEWqjQamy57h7kHOti02P2Dtf7ea4ks5qihWOgaJpZM4Jp1dm>
> .
>